### PR TITLE
[zh-cn] sync sidecar-containers

### PR DESCRIPTION
**Pull request by my-git9**
content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md

### CODE REVIEW

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 1]
```diff
@@ -15,34 +15,78 @@ weight: 50
...
 而无需直接修改主应用代码。
 
+<!--
+Typically, you only have one app container in a Pod. For example, if you have a web
+application that requires a local webserver, the local webserver is a sidecar and the
+web application itself is the app container.
+-->
+通常，一个 Pod 中只有一个应用程序容器。
+例如，如果你有一个需要本地 Web 服务器的 Web 应用程序，
+则本地 Web 服务器是 Sidecar，而 Web 应用程序本身是应用程序容器。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:33]:
```suggestion
则本地 Web 服务器以边车容器形式运行，而 Web 应用本身以应用容器形式运行。
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 2]
```diff
@@ -15,34 +15,78 @@ weight: 50
...
-Pod's `initContainers` field. These restartable _sidecar_ containers are independent with
-other [init containers](/docs/concepts/workloads/pods/init-containers/) and main
-application container within the same pod. These can be started, stopped, or restarted
-without affecting the main application container and other init containers.
+Kubernetes implements sidecar containers as a special case of
+[init containers](/docs/concepts/workloads/pods/init-containers/); sidecar containers remain
+running after Pod startup. This document uses the term _regular init containers_ to clearly
+refer to containers that only run during Pod startup.
+-->
+## Kubernetes 中的 Sidecar 容器   {#pod-sidecar-containers}
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:45]:
```suggestion
## Kubernetes 中的边车容器   {#pod-sidecar-containers}
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 3]
```diff
@@ -15,34 +15,78 @@ weight: 50
...
+
+Kubernetes 将 Sidecar 容器作为
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:47]:
```suggestion
Kubernetes 将边车容器作为
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 4]
```diff
@@ -15,34 +15,78 @@ weight: 50
...
+[Init 容器](/zh-cn/docs/concepts/workloads/pods/init-containers/)的一个特例来实现，
+Pod 启动后，Sidecar 容器仍保持运行。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:49]:
```suggestion
Pod 启动后，边车容器仍保持运行状态。
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 5]
```diff
@@ -15,34 +15,78 @@ weight: 50
...
+for containers listed in a Pod's `initContainers` field.
+These restartable _sidecar_ containers are independent from other init containers and from
+the main application container(s) within the same pod.
+These can be started, stopped, or restarted without effecting the main application container
+and other init containers.
 -->
-## 启用边车容器   {#enabling-sidecar-containers}
+如果你的集群启用了 `SidecarContainers`
+[特性门控](/zh-cn/docs/reference/command-line-tools-reference/feature-gates/)
+（该特性自 Kubernetes v1.29 起默认启动），你可以为 Pod 的 `initContainers`
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:64]:
```suggestion
（该特性自 Kubernetes v1.29 起默认启用），你可以为 Pod 的 `initContainers`
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 6]
```diff
@@ -15,34 +15,78 @@ weight: 50
...
+字段中列出的容器指定 `restartPolicy`。
+这些可重新启动的 **Sidecar** 容器独立于其他 Init 容器以及同一 Pod 内的主应用程序容器，
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:66]:
```suggestion
这些可重新启动的**边车（Sidecar）**容器独立于其他 Init 容器以及同一 Pod 内的主应用程序容器，
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 7]
```diff
@@ -15,34 +15,78 @@ weight: 50
...
-Pod 内的其他 [Init 容器](/zh-cn/docs/concepts/workloads/pods/init-containers/)及主应用容器相互独立。
-边车容器可以在不影响主应用容器和其他 Init 容器的情况下启动、停止或重启。
+<!--
+You can also run a Pod with multiple containers that are not marked as init or sidecar
+containers. This is appropriate if the containers within the Pod are required for the
+Pod to work overall, but you don't need to control which containers start or stop first.
+You could also do this if you need to support older versions of Kubernetes that don't
+support a container-level `restartPolicy` field.
+-->
+你还可以运行具有多个未标记为 Init 或 Sidecar 容器的 Pod。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:76]:
```suggestion
你还可以运行包含多个未标记为 Init 或边车容器的 Pod。
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 8]
```diff
@@ -15,34 +15,78 @@ weight: 50
...
+如果 Pod 内的容器需要 Pod 整体工作，那么这是合适的，但你不能要控制哪些容器首先启动或停止。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:77]:
```suggestion
如果作为一个整体而言，某个 Pod 中的所有容器都要运行，但你不需要控制哪些容器先启动或停止，那么这种设置是合适的。
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 9]
```diff
@@ -15,34 +15,78 @@ weight: 50
...
+如果你使用的是不支持容器级 `restartPolicy` 字段的旧版本 Kubernetes，你也可以这样做。
+
+<!--
+### Example application {#sidecar-example}
+
+Here's an example of a Deployment with two containers, one of which is a sidecar:
+-->
+### 应用示例   {#sidecar-example}
+
+下面是一个包含两个容器的 Deployment 示例，其中一个是 Sidecar：
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:87]:
```suggestion
下面是一个包含两个容器的 Deployment 示例，其中一个容器是边车形式：
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 10]
```diff
@@ -61,13 +105,13 @@ If a `readinessProbe` is specified for this init container, its result will be u
...
-ordering and sequential guarantees as other init containers, allowing them to
-be mixed with other init containers into complex Pod initialization flows.
+ordering and sequential guarantees as other init containers, allowing them to mix
+sidecar containers with regular init containers for complex Pod initialization flows.
 -->
 如果为此 Init 容器指定了 `readinessProbe`，其结果将用于确定 Pod 的 `ready` 状态。
 
 由于这些容器被定义为 Init 容器，所以它们享有与其他 Init 容器相同的顺序和按序执行保证，
-可以将它们与其他 Init 容器混合在一起，形成复杂的 Pod 初始化流程。
+允许它们将 Sidecar 容器与常规 Init 容器混合在一起，形成复杂的 Pod 初始化流程。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:114]:
```suggestion
从而允许将边车容器与常规 Init 容器混合使用，支持复杂的 Pod 初始化流程。
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 11]
```diff
@@ -86,45 +130,44 @@ kubelet 启动 `.spec.initContainers` 这一有序列表中的下一个 Init 容
 要么是其 `startupProbe` 成功而返回的结果。
 
 <!--
-Here's an example of a Deployment with two containers, one of which is a sidecar:
+### Jobs with sidecar containers
 -->
-以下是一个具有两个容器的 Deployment 示例，其中一个是边车：
-
-{{% code_sample language="yaml" file="application/deployment-sidecar.yaml" %}}
+### Sidecar 容器的工作
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:135]:
```suggestion
### 带边车容器的 Job {#jobs-with-sidecar-containers}
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 12]
```diff
@@ -86,45 +130,44 @@ kubelet 启动 `.spec.initContainers` 这一有序列表中的下一个 Init 容
...
-container will not prevent the Job from completing after the main container
-has finished.
+If you define a Job that uses sidecar using Kubernetes-style init containers,
+the sidecar container in each Pod does not prevent the Job from completing after the
+main container has finished.
 
 Here's an example of a Job with two containers, one of which is a sidecar:
 -->
-此特性也适用于带有边车的 Job，因为边车容器在主容器完成后不会阻止 Job 的完成。
+如果你使用 Kubernetes 样式的 Init 容器定义使用 Sidecar 的 Job，
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:144]:
```suggestion
如果你定义 Job 时使用基于 Kubernetes 风格 Init 容器的边车容器，
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 13]
```diff
@@ -86,45 +130,44 @@ kubelet 启动 `.spec.initContainers` 这一有序列表中的下一个 Init 容
...
+则每个 Pod 中的 Sidecar 容器不会阻止 Job 在主容器完成后完成。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:145]:
```suggestion
各个 Pod 中的边车容器不会阻止 Job 在主容器结束后进入完成状态。
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 14]
```diff
@@ -86,45 +130,44 @@ kubelet 启动 `.spec.initContainers` 这一有序列表中的下一个 Init 容
...
-## Differences from regular containers
+## Differences from application containers
 
-Sidecar containers run alongside regular containers in the same pod. However, they do not
+Sidecar containers run alongside _app containers_ in the same pod. However, they do not
 execute the primary application logic; instead, they provide supporting functionality to
 the main application.
 -->
-## 与常规容器的区别   {#differences-from-regular-containers}
+## 与应用程序容器的区别   {#differences-from-application-containers}
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:158]:
```suggestion
## 与应用容器间的区别   {#differences-from-application-containers}
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 15]
```diff
@@ -150,14 +193,21 @@ sidecar containers support [probes](/docs/concepts/workloads/pods/pod-lifecycle/
...
+
+Init containers stop before the main containers start up, so init containers cannot
+exchange messages with the app container in a Pod. Any data passing is one-way
+(for example, an init container can put information inside an `emptyDir` volume).
 
 ## Resource sharing within containers
 -->
-这些边车容器可以直接与主应用容器进行交互，共享相同的网络命名空间、文件系统和环境变量。
-所有这些容器紧密合作，提供额外的功能。
+Sidecar 容器可以直接与主应用程序容器交互，因为与 Init 容器一样，
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:206]:
```suggestion
边车容器可以直接与主应用容器交互，因为与 Init 容器一样，
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 16]
```diff
@@ -150,14 +193,21 @@ sidecar containers support [probes](/docs/concepts/workloads/pods/pod-lifecycle/
...
+它们始终共享相同的网络，并且还可以选择共享卷（文件系统）。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:207]:
```suggestion
它们总是与应用容器共享相同的网络，并且还可以选择共享卷（文件系统）。
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 17]
```diff
@@ -150,14 +193,21 @@ sidecar containers support [probes](/docs/concepts/workloads/pods/pod-lifecycle/
...
+
+Init 容器在主容器启动之前停止，因此 Init 容器无法与 Pod 中的应用程序容器交换消息。
+任何数据传递都是单向的（例如，Init 容器可以将信息放入 `emptyDir` 卷中）。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:210]:
```suggestion
所有数据传递都是单向的（例如，Init 容器可以将信息放入 `emptyDir` 卷中）。
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 18]
```diff
@@ -204,13 +254,18 @@ for resource usage apply:
...
-Pod 级别的 cgroup 是基于 Pod 的有效请求和限制值，与调度器相同。
+<!--
+### Sidecar containers and Linux cgroups {#cgroups}
+
+On Linux, resource allocations for Pod level control groups (cgroups) are based on the effective Pod
+request and limit, the same as the scheduler.
+-->
+### Sidecar 容器和 Linux Cgroup   {#cgroups}
+
+在 Linux 上，Pod Cgroup 的资源分配基于有效的 Pod 请求和限制，与调度程序相同。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:268]:
```suggestion
在 Linux 上，Pod Cgroup 的资源分配基于 Pod 级别的有效资源请求和限制，这一点与调度器相同。
```

File: content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md [hunk 19]
```diff
@@ -15,34 +15,78 @@ weight: 50
...
+the main application container(s) within the same pod.
+These can be started, stopped, or restarted without effecting the main application container
+and other init containers.
 -->
-## 启用边车容器   {#enabling-sidecar-containers}
+如果你的集群启用了 `SidecarContainers`
+[特性门控](/zh-cn/docs/reference/command-line-tools-reference/feature-gates/)
+（该特性自 Kubernetes v1.29 起默认启用），你可以为 Pod 的 `initContainers`
+字段中列出的容器指定 `restartPolicy`。
+这些可重新启动的**边车（Sidecar）**容器独立于其他 Init 容器以及同一 Pod 内的主应用程序容器，
```
Comment (mengjiao-liu) [inline content/zh-cn/docs/concepts/workloads/pods/sidecar-containers.md:66]: image
```suggestion
这些可重新启动的**边车（Sidecar）** 容器独立于其他 Init 容器以及同一 Pod 内的主应用程序容器，
```

**Review by mengjiao-liu — COMMENTED** [review]
/lgtm

**Pull request merged**